staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -141,22 +141,18 @@ func WithMaxInFlightLimit(
 
 			select {
 			case c <- true:
-				var mutatingLen, readOnlyLen int
-				if isMutatingRequest {
-					mutatingLen = len(mutatingChan)
-				} else {
-					readOnlyLen = len(nonMutatingChan)
-				}
-
 				defer func() {
 					<-c
+				}()
+
+				go func() {
```

Comment:
Spawning a goroutine for every request is not free. Since the mutating/readonly watermarks are increase-only, it's unclear to me that recording the watermark at the end of the request is a problem. Before taking on the cost of a goroutine like this, the problem with the current approach should be demonstrated.

Reply:
Thanks a lot, Got it.